processing/src/main/java/io/druid/segment/FloatDimensionHandler.java

```diff
@@ -0,0 +1,518 @@
+        return 1;
+      }
+      return Float.compare(o1, o2);
+    }
+  };
+
+  private final String dimensionName;
+
+  public FloatDimensionHandler(String dimensionName)
+  {
+    this.dimensionName = dimensionName;
+  }
+
+  @Override
+  public DimensionSelector getDimensionSelector(
+      ColumnSelectorFactory cursor, DimensionSpec dimSpec, ColumnCapabilities capabilities
+  )
+  {
+    return new FloatDimensionSelector(
+        cursor.makeFloatColumnSelector(dimensionName),
+        dimSpec.getExtractionFn(),
+        capabilities
+    );
+  }
+
+  @Override
+  public Function<Object, Float> getValueTypeTransformer()
+  {
+    return FLOAT_TRANSFORMER;
+  }
+
+  @Override
+  public Comparator<Float> getEncodedComparator()
+  {
+    return Float_COMPARATOR;
+  }
+
+  @Override
+  public Comparator<Float> getActualComparator()
+  {
+    return Float_COMPARATOR;
+  }
+
+  @Override
+  public byte[] getBytesFromRowValue(Float rowVal)
+  {
+    return Ints.toByteArray(Float.floatToIntBits(rowVal));
+  }
+
+  @Override
+  public Float getRowValueFromBytes(byte[] bytes)
+  {
+    return Float.intBitsToFloat(Ints.fromByteArray(bytes));
+  }
+
+  @Override
+  public int getEncodedValueSize()
+  {
+    return Floats.BYTES;
+  }
+
+  @Override
+  public DimensionIndexer makeIndexer(Object lock)
+  {
+    return new FloatDimensionIndexer();
+  }
+
+  @Override
+  public DimensionMerger makeMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      FileSmoosher v9Smoosher,
+      ObjectMapper mapper,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  )
+  {
+    return new FloatDimensionMerger(indexSpec, outDir, ioPeon, v9Smoosher, mapper, capabilities, progress);
+  }
+
+  @Override
+  public DimensionColumnReader makeColumnReader(Column column)
+  {
+    return new FloatDimensionColumnReader(column);
+  }
+
+
+  public class FloatDimensionIndexer implements DimensionIndexer<Float, Float>
+  {
+    private float minValue = Float.MAX_VALUE;
+    private float maxValue = Float.MIN_VALUE;
+
+    @Override
+    public Comparable[] processRowValsToIndexKey(Object dimValues)
+    {
+      final Comparable[] dimensionValues;
+      Float transformedVal = 0.0f;
+      try {
+        if (dimValues == null) {
+          //return new Comparable []{};
+          return null;
+        } else if (dimValues instanceof List) {
+          List<Object> dimValuesList = (List) dimValues;
+          dimensionValues = new Comparable[dimValuesList.size()];
+          for (int i = 0; i < dimValuesList.size(); i++) {
+            transformedVal = FLOAT_TRANSFORMER.apply(dimValuesList.get(i));
+            dimensionValues[i] = transformedVal;
+          }
+        } else {
+          transformedVal = FLOAT_TRANSFORMER.apply(dimValues);
+          dimensionValues = new Comparable[]{transformedVal};
+        }
+        //minValue = Float.compare(transformedVal, minValue) < 0 ? transformedVal : minValue;
```

Comment:
@fjy Removed min/max calculation for Long/Floats, added a note about that to DimensionIndexer interface